drizzlepac/hlautils/analyze.py

```diff
@@ -0,0 +1,265 @@
+    # available for the output table regardless of which keyword is used to 
+    # to determine the data is not viable for alignment.
+
+    for inputFile in inputFileList:
+
+        header_hdu  = 0
+        header_data = getheader(inputFile, header_hdu)
+
+        # Keywords to use potentially for downstream analysis
+        instrume = (header_data['INSTRUME']).upper()
+        detector = (header_data['DETECTOR']).upper()
+        subarray = header_data['SUBARRAY']
+        dateObs  = header_data['DATE-OBS']
+        mjdutc  = header_data['EXPSTART']
+
+        # Obtain keyword values for analysis of viability
+        obstype  = (header_data[OBSKEY]).upper()
+        mtflag   = (header_data[MTKEY]).upper()
+        
+        scan_typ = ''
+        if instrume == 'WFC3':
+            scan_typ = (header_data[SCNKEY]).upper()
+
+        sfilter = ''
+        if instrume == 'WFC3':
+            sfilter  = (header_data[FILKEY]).upper()
+        # Concatenate the two ACS filter names together with an underscore
+        # If the filter name is blank, skip it
+        if instrume == 'ACS':
+            for filtname in acsFiltNameList:
+
+                # The filter keyword value could be zero or more blank spaces 
+                # Strip off any leading or trailing blanks
+                if len(header_data[filtname].upper().strip()) > 0:
+
+                    # If the current filter variable already has some content,
+                    # need to append an underscore before adding more text
+                    if len(sfilter) > 0:
+                        sfilter += '_'
+                    sfilter += header_data[filtname].upper().strip()
+
+        aperture = (header_data[APKEY]).upper()
+        targname = (header_data[TARKEY]).upper()
+        exptime = header_data[EXPKEY]
+        fgslock = (header_data[FGSKEY]).upper()
+
+        chinject = 'NONE'
+        if instrume == 'WFC3' and detector == 'UVIS':
+            chinject = (header_data[CHINKEY]).upper()
+
+        # Determine if the image has one of these conditions.  The routine
+        # will exit processing upon the first satisfied condition.
+
+        noProcKey   = None
+        noProcValue = None
+        doProcess = True
+        # Imaging vs spectroscopic or coronagraphic
+        if obstype != 'IMAGING':
+            noProcKey   = OBSKEY
+            noProcValue = obstype 
+
+        # Moving target
+        elif mtflag == 'T':
+            noProcKey   = MTKEY
+            noProcValue = mtflag 
+
+        # Bostrophidon without or with dwell (WFC3 only)
+        elif any ([scan_typ == 'C', scan_typ == 'D']):
+            noProcKey   = SCNKEY
+            noProcValue = scan_typ
+
+        # Filter which does not begin with: 'F'(F###), 'C'(CLEAR), 'N'(N/A), and is not blank
+        # The sfilter variable may be the concatenation of two filters (F160_CLEAR)
+        elif sfilter[0] != 'F' and sfilter[0] != '' and sfilter[0] != 'C' and sfilter[0] != 'N': 
```

Comment:
@mdlpstsci A simpler and more readable version: `elif sfilter and not sfilter.startswith(('F', 'C', 'N')):` or `elif sfilter and sfilter[0] not in ['F', 'C', 'N']:`. Also, if `sfilter` can be an empty string then the original code will crash when attempting to get the first character `sfilter[0]`.